restorer.go

```diff
@@ -40,6 +41,8 @@ func (r *Restorer) Restore(cache Cache) error {
 	}
 
 	var g errgroup.Group
+	umask := archive.SetUmask(0)
+	defer archive.SetUmask(umask)
```

Comment:
I'm not quite sure what these lines will prevent. Ultimately, the same `archive.Extract` code will be called.

Reply:
Replied [here](https://github.com/buildpacks/lifecycle/pull/722/files#r715649087)